pkg/plugin/plugin.go

```diff
@@ -145,23 +193,70 @@ func (r *RpcPlugin) VerifyWeight(rollout *v1alpha1.Rollout, desiredWeight int32,
 }
 
 func (r *RpcPlugin) RemoveManagedRoutes(rollout *v1alpha1.Rollout) pluginTypes.RpcError {
-	// we could remove the canary destination, but not required since it will have 0 weight at the end of rollout
-	return pluginTypes.RpcError{}
-}
-
-func (r *RpcPlugin) Type() string {
-	return Type
-}
+	if !slices.ContainsFunc(rollout.Spec.Strategy.Canary.Steps, func(s v1alpha1.CanaryStep) bool {
+		return s.SetHeaderRoute != nil
+	}) {
+		// none of the steps have a SetHeaderRoute so nothing to clean up
+		return pluginTypes.RpcError{}
+	}
 
-func getPluginConfig(rollout *v1alpha1.Rollout) (*GlooPlatformAPITrafficRouting, error) {
-	glooplatformConfig := GlooPlatformAPITrafficRouting{}
+	ctx := context.TODO()
+	glooPluginConfig, err := getPluginConfig(rollout)
+	if err != nil {
+		return pluginTypes.RpcError{
+			ErrorString: err.Error(),
+		}
+	}
 
-	err := json.Unmarshal(rollout.Spec.Strategy.Canary.TrafficRouting.Plugins[PluginName], &glooplatformConfig)
+	// get the matched routetables
+	matchedRts, err := r.getRouteTables(ctx, rollout, glooPluginConfig)
 	if err != nil {
-		return nil, err
+		return pluginTypes.RpcError{
+			ErrorString: err.Error(),
+		}
+	}
+	if len(matchedRts) == 0 {
+		// nothing to update, don't bother computing things
+		return pluginTypes.RpcError{}
 	}
 
-	return &glooplatformConfig, nil
+	var combinedError error
+	for _, rt := range matchedRts {
+		originalRouteTable := &networkv2.RouteTable{}
+		rt.RouteTable.DeepCopyInto(originalRouteTable)
+		newRoutes := slices.DeleteFunc(rt.RouteTable.Spec.Http, func(r *networkv2.HTTPRoute) bool {
+			for _, managed := range rollout.Spec.Strategy.Canary.TrafficRouting.ManagedRoutes {
+				if strings.EqualFold(r.GetName(), managed.Name) {
+					return true
+				}
+			}
+			return false
+		})
+
+		rt.RouteTable.Spec.Http = newRoutes
+		if !r.IsTest {
```

Comment:
Even if you don't remove this field (which I think we should remove the field if we have the time), we could early return to flatten out these statements

```
if r.isTest {
     r.LogCtx.Debugf("test route table http routes: %v", rt.RouteTable.Spec.Http)
     continue
}

if e := patchRouteTable(ctx, r.Client, rt.RouteTable, originalRouteTable)l; err != nil {
     combinedError = errors.Join(combinedError, e)
     continue
}

r.LogCtx.Debugf("patched route table %s.%s", rt.RouteTable.Namespace, rt.RouteTable.Name)
```